From 296863b6902f6b005a0d643bc2914517a1508f62 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Thu, 30 May 2019 11:44:47 +0200
Subject: [PATCH 267/386] arm64: dts: marvell: fix switch gpio description on
 Cybertan board

The Cybertan Board 88E6390 switch has a gpio interrupt line
connected to the SoC. However it failed to register it in
a kernel with a following warning:

"irq: type mismatch, failed to map hwirq-30 for \
/cp1/config-space/system-controller@440000/gpio@100!"

Fix that by using falling edge interrupt type, which is
accepted by the CP0 GPIO controller.

Change-Id: I1f2a15e7c2ac602c4a2bd48c01290beaadb3d907
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/10231
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-ucpe.dts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-ucpe.dts b/arch/arm64/boot/dts/marvell/armada-8040-ucpe.dts
index 90c5fd9cf5d1..3f61d4da62d0 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-ucpe.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-ucpe.dts
@@ -224,7 +224,7 @@
 		eeprom-length = <65536>;
 
 		interrupt-parent = <&cp1_gpio1>;
-		interrupts = <30 IRQ_TYPE_LEVEL_LOW>;
+		interrupts = <30 IRQ_TYPE_EDGE_FALLING>;
 		interrupt-controller;
 		#interrupt-cells = <2>;
 
-- 
2.17.1

